File: example/python/demo_torque_control.py
import time
import fi_fsa_v3

def main():
    print(f"Hello, FSA!\n")
    device = fi_fsa_v3.FSA()
    ret = device.Init("192.168.137.101")
    if ret != fi_fsa_v3.ret_e.SUCCESS:
        print(f"Init returned: {ret}")
        exit(1)
    print(f"FSA initialized\n")

    ret = device.EnableControl( fi_fsa_v3.ctrl_mode_e.TORQUE_MODE, 1000, 0 )
    if ret != fi_fsa_v3.ret_e.SUCCESS:
        print(f"FSA Enable Control failed with code: {ret}")

    tor = 0
    for size_t in range(30):
        tor += 0.1
        ret = device.SetTorque(tor)
        if ret != fi_fsa_v3.ret_e.SUCCESS:
            print(f"FSA Set Torque failed with code: {ret}")
        time.sleep(0.2)

    for size_t in range(30, 0, -1):
        tor -= 0.1
        ret = device.SetTorque(tor)
        if ret != fi_fsa_v3.ret_e.SUCCESS:
            print(f"FSA Set Torque failed with code: {ret}")
        time.sleep(0.2)

    ret = device.DisableControl()
    if ret != fi_fsa_v3.ret_e.SUCCESS:
        print(f"FSA Disable Control failed with code: {ret}")

if __name__ == "__main__":
    main()

File: example/cpp/demo_torque_control.cpp
#include <stdio.h>
#include <thread>

#include "fsav3.h"
using namespace AC3;

int main() {
    printf("Hello, FSA!\n");
    FSA        fsa;
    FSA::ret_e ret;
    ret = fsa.Init( "192.168.137.101" );
    if ( ret != FSA::ret_e::SUCCESS ) {
       printf( "FSA Init failed, ret: %d\n", ( int )ret );
       return 0;
    }
    printf( "FSA initialized\n" );
    
    ret = fsa.EnableControl( FSA::ctrl_mode_e::TORQUE_MODE, 1000, 0 );
    if ( ret != FSA::ret_e::SUCCESS ) {
        printf( "FSA Enable Control failed with code: %d\n", ( int )ret );
    }

    double tor = 0;
    for (size_t i = 0; i < 30; i++) {
        tor += 0.1;
        ret = fsa.SetTorque(tor);
        if ( ret != FSA::ret_e::SUCCESS ) {
            printf("FSA Set Torque failed with code: %d\n", ( int )ret );
        }
        std::this_thread::sleep_for( std::chrono::microseconds( 200000 ) );
    }

    for (size_t i = 30; i > 0; i--) {
        tor -= 0.1;
        ret = fsa.SetTorque(tor);
        if ( ret != FSA::ret_e::SUCCESS ) {
        printf("FSA Set Torque failed with code: %d\n", ( int )ret );
        }
        std::this_thread::sleep_for( std::chrono::microseconds( 200000 ) );
    }

    ret = fsa.DisableControl();
    if ( ret != FSA::ret_e::SUCCESS ) {
        printf("FSA Disable Control failed with code: %d\n", ( int )ret );
    }

    return 0;
}